### DISCUSSION TITLE
Why is the 6522 inside the RIA address space ?

### DISCUSSION THREAD

**Discussion by GlennFR** [picocomputer/community#117] [Q&A] [answered]
This question has been worrying me for quite a while...  It's actually the RIA that's selecting the 6522 VIA when the FFDx address range appears on the address bus.  Is the RIA actually going to use the VIA for something?  Is the VIA used, for example, if there is no pico VGA on the board ?
My real question is : if I wish to move the VIA below the FFxx watermark, and add other H/W devices (e.g. for I2C/SPI) will the RIA still work OK?  I couldn't see any code pertaining to the VIA in the 0.8 sources.
Thanks for any insight.

**Comment by ceptimus** [comment]
I thought the RIA was at $FFE0-$FFFF, so the VIA isn't in the RIA's address space?  And the two chip select lines for the 6522 VIA come from the address decode logic (the 74-series chips) not from the RIA? I didn't think the RIA could directly read or write the 6522, even if it wanted to: the 6522 is only directly accessible to the 6502.

**Reply by GlennFR** [reply]
Well, yes and no... According to the schematic diagram, the 74xx logic actually decodes $FFEx (address lines A15..8 = FFxx, plus A7, A6, A5 = Ex)
To decode $xxDx requires address bit A4 - which doesn't go to any of the 74xx devices.  A4 *does* go to the RIA, however, - and it also has the RREQ line connected to GP6.  So it looks like it's the RIA that does the CS logic for the VIA...  But I may have missed something else.

**Reply by ceptimus** [reply]
Yes, I see what you mean.  I studied the schematic for a while, and now I'm confused too.

**Comment by rumbledethumps** [comment]
The 74XX logic handles chip selections. This is covered in a video... **https://youtu.be/wxV6x5BUMH4**

**Reply by rumbledethumps** [reply, marked as answer]
Oh, I see. You think $FFDx needs to be decoded. But it doesn't. The reserved space is filled with shadows of the VIA because that made the reference design cheaper and easier.

**Reply by GlennFR** [reply]
Hi Rumble, thanks for clearing the question marks.  So is it safe to completely remove the VIA, then ?  The RIA won't complain?

**Reply by rumbledethumps** [reply]
Removing the VIA from the reference design will leave a floating IRQ input. The 6502 will complain loudly.

**Reply by ceptimus** [reply]
If you want to run without a 6522, I suppose you could add a link wire, or perhaps better a 10k resistor, to the chip socket, (or PCB if you've not fitted a socket) to pull the IRQ pin permanently high.
